pkg/k8s/client.go

```diff
@@ -150,3 +150,18 @@ func buildSmiSplitClient(log logrus.FieldLogger, config *rest.Config) (*splitcli
 
 	return client, nil
 }
+
+// NewKubernetesClient creates a new Kubernetes client.
+func NewKubernetesClient(log logrus.FieldLogger, masterURL, kubeConfig string) (*kubernetes.Clientset, error) {
```

Comment:
For now, I would just keep the `NewClient` method and we will revisit this part later.